lib/overcommit/git_repo.rb

```diff
@@ -273,5 +273,33 @@ def branches_containing_commit(commit_ref)
     def current_branch
       `git symbolic-ref --short -q HEAD`.chomp
     end
+
+    # Returns current branch's tracking branch name when it is available
+    # or its parent branch.
+    #
+    # @return [String]
+    def pre_push_ref
+      if tracking_branch_ref.empty?
+        parent_branch
+      else
+        tracking_branch_ref
+      end
+    end
+
+    private_class_method
+
+    def tracking_branch_ref
+      `git rev-parse \
+      --symbolic-full-name \
+      --abbrev-ref @{u} \
+      2> #{File::NULL}`.chomp
+    end
+
+    def parent_branch
+      `git show-branch -a \
```

Comment:
On a local repo of mine with a lot of remote branches, this can display warnings like:

```
warning: ignoring origin/some-remote-branch; cannot handle more than 25 refs
```

Would need to filter out stderr to avoid.

Reply:
Nice catch. Will update accordingly.